### PR TITLE
Add routing functionality

### PR DESCRIPTION
**Pull request by daniyalnagori**
The requirements are done.Filter Categories all unactive properties are disabled while they are not active But filterText has initial value empty string.So this means its not disabled while from its initial state that's why it will show "filterText=&" if you want it to be disabled also while has no value so we can make it conditional or set its initial value from its initial state from empty string to null so, let me know.

### CODE REVIEW

**Comment by mmmatthew** [comment]
comment: this version of the app works with datablue version https://github.com/water-fountains/datablue/commit/45171e688c1acf2d38f052f2913df24bd8abc6fd

**Comment by mmmatthew** [comment]
Must be fixed:
- changing language in app should update url
- bug: 
  - reload the page
  - select a fountain >> the map zooms in
  - close the fountain detail
  - select another fountain >> the map doesn't zoom in
- similarly, when I select a fountain and copy paste the url, the map doesn't go into zoomed-in mode.

Does not need to be fixed:
- [ ] changing filter, e.g. clicking 'only spring water' is not reflected in url 
  --> Just omit the filters from the url parameters for now

**Comment by daniyalnagori** [comment]
Please check the latest changes and notify me if you find anything left.

File: src/app/data.service.ts [hunk 1]
```diff
@@ -160,17 +163,29 @@ export class DataService {
...
+        (data: FeatureCollection<any>) => {
+          this.ngRedux.dispatch({ type: GET_DIRECTIONS_SUCCESS, payload: data });
           this.directionsLoadedSuccess.emit(data);
         });
   }
 
+  // Fountain selector checking
+  checkFountainSelector(): any {
+    try {
+      let url = `https://water-fountains.org:3000/api/v1/fountain?queryType=byCoords&database=wikidata&idval=Q27229889&lat=47.364622&lng=8.537836`;
```
Comment (mmmatthew) [inline src/app/data.service.ts:175]: The fountain selector is hardcoded - this is not a real check.

File: src/app/list/list.component.ts [hunk 2]
```diff
@@ -12,50 +15,61 @@ import {HIGHLIGHT_FOUNTAIN, SELECT_FOUNTAIN} from '../actions';
 export class ListComponent implements OnInit {
   public fountains = [];
 
-  constructor(public dataService: DataService, private ngRedux: NgRedux<IAppState>) {
+  constructor(public dataService: DataService, private ngRedux: NgRedux<IAppState>,
+    public route: RouteCheckerService, public translate: TranslateService) {
```
Comment (mmmatthew) [inline src/app/list/list.component.ts:19]: "List component" just displays the list of fountains. This is not the place for route checking logic.

**Comment by mmmatthew** [comment]
Which commit of datablue are you working with?

File: src/app/route-checker.service.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    else if (this._basel.test(location.pathname) || location.pathname == "/b%C3%A2le") {
+      this.checkFilterCategories("basel", lang);
+      this.cityName = "basel"
+    }
+    else {
+      this.checkFilterCategories("zurich", lang);
+      this.cityName = "zurich"
+
+    }
+    this.ngRedux.dispatch({ type: UPD_CITY, payload: this.cityName })
```
Comment (mmmatthew) [inline src/app/route-checker.service.ts:52]: should only fire if the city is changed. here it is fired every time the route is checked.

File: src/app/navbar/navbar.component.ts [hunk 4]
```diff
@@ -15,32 +316,92 @@ export class NavbarComponent implements OnInit {
...
+    })
   }
 
   ngOnInit() {
+    if (this.lang !== this.langOpted) {
+      this.ngRedux.dispatch({ type: CHANGE_LANG, payload: this.langOpted })
+    }
+    var urlParams = new URLSearchParams(window.location.search);
+    var url = new URL(window.location.href);
+    var c = url.searchParams.get("idval");
```
Comment (mmmatthew) [inline src/app/navbar/navbar.component.ts:352]: navbar is not the right location to put this logic.

**Comment by mmmatthew** [comment]
Functionality has been integrated into project.

**Pull request closed without merging**